tensorflow/security/fuzzing/core/framework/function_fuzz.cc

```diff
@@ -0,0 +1,148 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <vector>
+
+#include <gtest/gtest.h>
+#include "fuzztest/fuzztest.h"
+#include "tensorflow/security/fuzzing/domains/function_def.h"
+
+namespace tensorflow {
+namespace fuzzing {
+
+using AttrValueVariantFuzzType = std::variant<std::string, int, float, std::vector<std::string>>;
+using AttrsFuzzType = std::vector<std::pair<std::string, AttrValueVariantFuzzType>>;
+using AttrsVectorType = std::vector<std::pair<std::string, FunctionDefHelper::AttrValueWrapper>>;
+
+namespace domain {
+
+AttrsVectorType AttrsVectorGenerator(AttrsFuzzType attrsFuzz) {
+  AttrsVectorType attrs;
+  attrs.reserve(attrsFuzz.size());
+  std::transform(attrsFuzz.cbegin(), attrsFuzz.cend(), std::back_inserter(attrs),
+                 [](std::pair<std::string, AttrValueVariantFuzzType> attr) {
+                   return std::visit([&](auto&& value) {
+                     return std::make_pair(attr.first, FunctionDefHelper::AttrValueWrapper{value});
+                   }, attr.second);
+                 });
+  return attrs;
+}
+
+auto Attrs() {
+  return fuzztest::Map(AttrsVectorGenerator, fuzztest::Arbitrary<AttrsFuzzType>());
+}
+
+auto RandomNode() {
```

Comment:
Should probably avoid using the word Random here, since we are generating a domain. Maybe FunctionDefNodeDomain since Node is also a very overloaded term? 

Also, you already have a domain subfolder, maybe this should go there?

Reply:
I changed the `RandomNode` to `FunctionDefNode`. I skipped the `Domain` suffix because it is used as `domain::FunctionDefNode`